container/container.go

```diff
@@ -449,17 +449,23 @@ func (w *ContainerWorker) finalizeDeployment(agreementId string, deployment *con
 		// the cgoup permission can be omitted. It defaults to "rwm" when omitted.
 		for _, givenDevice := range service.Devices {
 			cgp := "rwm"
+			pic := ""
 			sp := strings.Split(givenDevice, ":")
 			if len(sp) == 3 {
 				// the cgroup permission
 				cgp = sp[2]
-			} else if len(sp) < 2 || len(sp) > 3 {
+				pic = sp[1]
+			} else if len(sp) == 2 {
+				pic = sp[1]
+			}else if len(sp) < 1{
+				pic = sp[0]
+			} else if len(sp) <= 0 {
```

Comment:
above line should be `else if len(sp) <= 0 || len(sp) > 3 {`

Reply:
Oh yes. I understand the issue. Sorry I did not catch this!